modules/apps/static/portal-osgi-web/portal-osgi-web-wab-generator-impl/src/main/java/com/liferay/portal/osgi/web/wab/generator/internal/cdi/CDIAnnotationReader.java

```diff
@@ -0,0 +1,763 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.osgi.web.wab.generator.internal.cdi;
+
+import aQute.bnd.component.annotations.ReferenceCardinality;
+import aQute.bnd.osgi.Analyzer;
+import aQute.bnd.osgi.Annotation;
+import aQute.bnd.osgi.ClassDataCollector;
+import aQute.bnd.osgi.Clazz;
+import aQute.bnd.osgi.Descriptors;
+import aQute.bnd.osgi.Instruction;
+import aQute.bnd.signatures.ClassResolver;
+import aQute.bnd.signatures.ClassSignature;
+import aQute.bnd.signatures.ClassTypeSignature;
+import aQute.bnd.signatures.FieldResolver;
+import aQute.bnd.signatures.FieldSignature;
+import aQute.bnd.signatures.JavaTypeSignature;
+import aQute.bnd.signatures.MethodResolver;
+import aQute.bnd.signatures.MethodSignature;
+import aQute.bnd.signatures.ReferenceTypeSignature;
+import aQute.bnd.signatures.Result;
+import aQute.bnd.signatures.VoidDescriptor;
+import aQute.bnd.version.Version;
+
+import java.lang.annotation.ElementType;
+
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * @author Raymond Augé
+ */
+public class CDIAnnotationReader extends ClassDataCollector {
+
+	public static final Version CDI_ARCHIVE_VERSION = new Version(1, 1, 0);
+
+	public static final Version V1_0 = new Version(1, 0, 0);
+
+	public static List<BeanDef> getDefinition(
+			Clazz c, Analyzer analyzer, EnumSet<Discover> options)
+		throws Exception {
+
+		CDIAnnotationReader r = new CDIAnnotationReader(analyzer, c, options);
+
+		return r._defs();
+	}
+
+	@Override
+	public void annotation(Annotation annotation) {
+		try {
+			Descriptors.TypeRef typeRef = annotation.getName();
+
+			String fqn = typeRef.getFQN();
+
+			if ("org.osgi.service.cdi.annotations.Bean".equals(fqn)) {
```

Comment:
@rotty3000 for example, we are not hitting any of these if conditions by deploying the testing wars. And many other branches in this class and CDIAnnotations too.

Is there a real expected use case for these?

Reply:
sigh.... I'm going to take a different approach.